web-common/src/features/dashboards/state-managers/selectors/measure-filters.ts

```diff
@@ -85,37 +85,59 @@ export const getMeasureFilterItems = (
     const addedMeasure = new Set<string>();
 
     for (const dtf of dashData.dashboard.dimensionThresholdFilters) {
-      forEachExpression(dtf.filter, (e, depth) => {
-        if (depth > 0 || !e.cond?.exprs?.length) {
-          return;
-        }
-        const ident =
-          e.cond?.exprs?.[0].ident ?? e.cond?.exprs?.[0].cond?.exprs?.[0].ident;
-        if (
-          ident === undefined ||
-          addedMeasure.has(ident) ||
-          !measureIdMap.has(ident)
-        ) {
-          return;
-        }
-        const measure = measureIdMap.get(ident);
-        if (!measure) {
-          return;
-        }
-        addedMeasure.add(ident);
-        filteredMeasures.push({
-          dimensionName: dtf.name,
-          name: ident,
-          label: measure.label || measure.expression || ident,
-          expr: e,
-        });
-      });
+      filteredMeasures.push(
+        ...getMeasureFilterItemsForFilter(
+          measureIdMap,
+          dtf.filter,
+          dtf.name,
+          addedMeasure,
+        ),
+      );
     }
 
     return filteredMeasures;
   };
 };
 
+export function getMeasureFilterItemsForFilter(
```

Comment:
Similarly, maybe `getMeasureFilters()` would be more readable?

Reply:
Though I see there's already a `getMeasureFilters()` function in `features/dashboards/filters/measure-filters/measure-filter-utils.ts`. Do we need both? Is it that the function below is a small transformation on top of the existing function?

Reply:
They are all slightly different. Makes sense to rename the one in `measure-filter-utils.ts`